### PR TITLE
Separated out working tests from not working tests for NNC OpInfo

### PR DESCRIPTION
**Pull request by Chillee**
This gets rid of a lot of the try/else rigamarole.

### CODE REVIEW

File: test/test_jit_fuser_te.py [hunk 1]
```diff
@@ -1808,129 +1808,171 @@ def eager(x, y, z: int):
...
-                # output
-                if isinstance(out, tuple):
-                    continue
+            g = {'torch': torch, 'inf' : math.inf, 'op': op}
+            exec(code, g)
+            f = g['f']
+            f.__module__ = 'test'
+            out = f(*param_values)
+
+            # NNC currently oftens segfault when asked to lower ops with 0-dim tensor outputs
```
Comment (ZolotukhinM) [inline test/test_jit_fuser_te.py:1931]: Maybe we should add such ops to known failures?

File: test/test_jit_fuser_te.py [hunk 2]
```diff
@@ -1808,129 +1808,171 @@ def eager(x, y, z: int):
...
-                    continue
-                ts_g = torch.jit.trace(f, param_values)
-                kernel = torch._C._te.TensorExprKernel(ts_g.graph)
-                self.assertEqual(kernel.run(tuple(param_values)), f(*param_values))
-                self.assertEqual(kernel.fallback(tuple(param_values)), f(*param_values))
+            ts_g = torch.jit.trace(f, param_values)
+            kernel = torch._C._te.TensorExprKernel(ts_g.graph)
+            correct_val = f(*param_values)
+            self.assertEqual(kernel.run(tuple(param_values)), correct_val)
+            self.assertEqual(kernel.fallback(tuple(param_values)), correct_val)
```
Comment (ZolotukhinM) [inline test/test_jit_fuser_te.py:1941]: Running fallback is probably unnecessary - it should simply interpret the graph. But I don't mind leaving it here.

File: test/test_jit_fuser_te.py [hunk 3]
```diff
@@ -1808,129 +1808,171 @@ def eager(x, y, z: int):
...
+
+        if not is_compiling:
+            raise RuntimeError("Skipped all inputs")
```
Comment (ZolotukhinM) [inline test/test_jit_fuser_te.py:1944]: What's this for? Should it be an error?

**Review by ZolotukhinM — APPROVED** [review]
👍

File: test/test_jit_fuser_te.py (hunk 1 above)
Comment (Chillee) [reply test/test_jit_fuser_te.py:1931]: I considered that, but the problem is that it happens for most ops lol. Most ops (including unary ops like `torch.sin`) can operator on scalars.

File: test/test_jit_fuser_te.py (hunk 3 above)
Comment (Chillee) [reply test/test_jit_fuser_te.py:1944]: This is just to keep track of operations that always output scalars (like `torch.trace`). So, if all the inputs always result in scalar outputs, then this operation never gets tested, and i thought it'd be strange to put those ops in the "works" list.

Comment (Chillee) [reply test/test_jit_fuser_te.py:1944]: I'll add a comment about it.

**Pull request closed without merging**